connect_labs/labs/indicators/sources/geotiff.py

```
@@ -164,23 +306,32 @@ def one(tag: int, default=None):
+        """One strip or tile, decompressed, un-predicted, as (rows, cols, samples)."""
+        raw = _decompress(buf[offset : offset + length], compression)
+        # A copy, not a view: undoing the predictor writes in place, and a
+        # buffer read straight out of the file is read-only.
+        arr = np.frombuffer(raw, dtype=np_dtype, count=rows * cols * samples).reshape(rows, cols, samples).copy()
+        return _unpredict(arr, predictor, samples, bo)
+
     out = np.zeros((height, width), dtype=np_dtype)
     if _TILE_OFFSETS in tags:
         tw, th = one(_TILE_WIDTH), one(_TILE_LENGTH)
         across = (width + tw - 1) // tw
         for i, (offset, length) in enumerate(zip(tags[_TILE_OFFSETS], tags[_TILE_BYTE_COUNTS])):
```

Ruff (0.16.2)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)